k8s/upgrades/0.8.0-0.8.1/cstor_pool_update.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+##################################################################
+# STEP: Get SPC name as argument                                 #
+#                                                                #
+# NOTES: Obtain the pool deployments to perform upgrade operation#
+##################################################################
+
+function usage() {
+    echo
+    echo "Usage:"
+    echo
+    echo "$0 <spc-name> <openebs-namespace>"
+    echo
+    echo "  <spc-name> Get the SPC name using: kubectl get spc"
+    echo "  <openebs-namespace> Get the namespace where openebs"
+    echo "    pods are installed"
+    exit 1
+}
+
+function setDeploymentRecreateStrategy() {
+    ns=$1
+    dn=$2
+    currStrategy=`kubectl get deploy -n $ns $dn -o jsonpath="{.spec.strategy.type}"`
+
+    if [ $currStrategy = "RollingUpdate" ]; then
+       kubectl patch deployment --namespace $ns --type json $dn -p "$(cat patch-strategy-recreate.json)"
+       rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+       echo "Deployment upgrade strategy set as recreate"
+    else
+       echo "Deployment upgrade strategy was already set as recreate"
+    fi
+}
+
+##Checking the version of OpenEBS using image in deployment ####
+function checkVersion() {
+	ns=$1
+	check_deply=$2
+	openebs_version=$(kubectl get deploy $check_deply -n $ns\
+        -o jsonpath="{.spec.template.spec.containers[0].image}" | cut -d ':' -f 2)
+
+	if [ $openebs_version != "0.8.0" ]; then
+    		echo "Current replica version is not 0.8.0";exit 1;
+	fi
+
+}
+
+if [ "$#" -ne 2 ]; then
+    usage
+fi
+
+spc=$1
+ns=$2
+pool_cnt=0
+
+# Get the list of pool deployments for given SPC, delimited by ':'
+pool_deploys=`kubectl get deploy -n $ns \
+ -l app=cstor-pool,openebs.io/storage-pool-claim=$spc \
+ -o jsonpath="{range .items[*]}{@.metadata.name}:{end}"`
+
+echo "Patching Pool Deployment upgrade strategy as recreate"
+for pool_dep in `echo $pool_deploys | tr ":" " "`; do
+    if [ $pool_cnt -eq 0 ]; then
+    	checkVersion $ns $pool_dep
+    fi
+    setDeploymentRecreateStrategy $ns $pool_dep
```

Comment:
why do we need to set this?

Reply:
I mean - why do we need to set this? What is the pool deployment 'Strategy' in 0.8?

Reply:
pool deployment strategy is set to `Recreate` in 0.8.0. If we set to Recreate then patching will happen by terminate the old version and release the new one.

Reply:
Having a check will make sure that stratagey is set to Recreate.